lib/internal/Magento/Framework/ObjectManager/Config/Config.php

```diff
@@ -152,10 +152,17 @@ public function isShared($type)
      *
      * @param string $instanceName
      * @return mixed
+     * @throws \LogicException
      */
     public function getInstanceType($instanceName)
     {
         while (isset($this->_virtualTypes[$instanceName])) {
+            if ($instanceName == $this->_virtualTypes[$instanceName]) {
```

Comment:
Also, can we make this a strict comparison? `===` instead of `==`? Or will that miss some specific cases?